### ISSUE TITLE
Proposal: new config option to enable/disable this plugin for `mkdocs serve`

### ISSUE THREAD

**Issue by tombreit** [timvink/mkdocs-git-authors-plugin#80]
We now use three mkdocs-git-*-plugins and the execution time is greatly increased by these plugins. This is particularly noticeable during `mkdocs serve`. 

I discovered the `enabled_on_serve` option at @squidfunk [info plugin](https://github.com/squidfunk/mkdocs-material/blob/3e862b5e992e934a033413263f60ab0e95ed209f/src/plugins/info/config.py#L31)  and implemented this in (my) plugin:

https://github.com/tombreit/mkdocs-git-latest-changes-plugin/commit/f467fec88ec55d34eeacf9a525adddd334dbb602

Would this also be something for mkdocs-git-authors-plugin?

**Comment by timvink** [comment]
I like the idea! Would you be open to contributing a PR?

There are quite some git plugins out there. Ideally, you would only access git information once. I started with the idea for a new `mkdocs-git-info-plugin` that would serve as an information source for plugins that need git info, but I haven't made the time to complete it. Optimizations like using multiple threads could also take place there.

**Comment by squidfunk** [comment]
> There are quite some git plugins out there. Ideally, you would only access git information once. I started with the idea for a new mkdocs-git-info-plugin that would serve as an information source for plugins that need git info, but I haven't made the time to complete it. Optimizations like using multiple threads could also take place there.

Yes, yes, yes! One plugin to rule them all, that would be amazing and consolidate efforts big time. We'd switch to it in an instant once it supports the functionality we integrate with in Material for MkDocs ✌️

**Comment by tombreit** [comment]
Hi  @timvink , would you also like this functionality for mkdocs-git-revision-date-localized-plugin?

**Comment by timvink** [comment]
Just released the new version to pypi, thanks again! 

> Hi @timvink , would you also like this functionality for mkdocs-git-revision-date-localized-plugin?

Short answer is yes, go for it!

Long answer is maybe we can do more.. there are many plugins that might want to use this. Have you seen the `groups` plugin? https://squidfunk.github.io/mkdocs-material/plugins/group/ . Perhaps the right approach is to add a `enabled_on_serve` parameter there, or just use the env var?

I also feel that `enabled` and `enabled_on_serve` are parameters that every mkdocs plugin should have. If you want to push this topic, you could open an issue on the mkdocs project and we discuss it there?

**Comment by squidfunk** [comment]
@timvink at least `enabled` should land pretty soon, see https://github.com/mkdocs/mkdocs/issues/3386.
